packit/patches.py

```diff
@@ -26,17 +26,73 @@
 import logging
 from itertools import islice
 from pathlib import Path
-from typing import List, Tuple, Optional
+from typing import List, Optional
 
 import git
+import yaml
 
 from packit.constants import DATETIME_FORMAT
+from packit.git_utils import get_metadata_from_message
 from packit.local_project import LocalProject
 from packit.utils import is_a_git_ref, run_command
 
 logger = logging.getLogger(__name__)
 
 
+class PatchMetadata:
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        path: Optional[Path] = None,
+        location_in_specfile: Optional[str] = None,
+        description: Optional[str] = None,
+        commit: Optional[git.Commit] = None,
```

Comment:
let's do a list of commits please so people can do fixups easily

Reply:
Good point. Where we want to handle this, in git or in our code?

If `git`, it will require a different patch creation but why not. We are creating all at once without considering any fixup comments.

Does this consider fix-ups?
```python
            git_f_p_cmd = [
                "git",
                "format-patch",
                "--output-directory",
                f"{destination}",
                git_ref,
                "--",
                ".",
            ] + [f":(exclude){file_to_ignore}" for file_to_ignore in files_to_ignore]
```

Or, we can go through `commit.summary` and try to do fix-up ourselves but I would try to use `git` if possible.

Reply:
This is the solution I had in mind (pseudocode):
```
$ git checkout -B temp-branch
$ for commit in patch.commits: git cherry-pick $commit; done
$ git format-patch  # from the temp-branch and create only a single patch file
```

Reply:
in the initial implmentation, I'd only implement it for a single commit and do multiple commits in future

Reply:
Ok, so handle this ourselves.

I originally thought about using `--autosquash` with rebase to handle this in git and don't need to check fixup commits.

> This is the solution I had in mind (pseudocode)

Yes, that looks good.

Reply:
@TomasTomecek I'll leave it as is for now:

- This does not change the format for the metadata.
- There are multiple approaches to this. (There is also some hack to be able to use `autosquash`.)
  - I can present and compare various ways for that and implement the best one separately.

Is it ok for you?

Reply:
yes, fine with me, we can improve in future